extensions/io/org.eclipse.smarthome.io.mqttembeddedbroker/src/main/java/org/eclipse/smarthome/io/mqttembeddedbroker/internal/EmbeddedBrokerService.java

```diff
@@ -161,130 +33,16 @@ public void brokerRemoved(String brokerID, MqttBrokerConnection broker) {
-        // TODO wait for NetworkServerTls implementation
-        // try {
-        // final SSLContext sslContext = networkServerTls.createSSLContext("mqtt");
-        // server.startServer(new MemoryConfig(properties), null, () -> sslContext, authentificator, authorizer);
-        // } catch (GeneralSecurityException | IOException e) {
-        // logger.error("No SSL available", e);
-        server.startServer(new MemoryConfig(properties), null, null, authentificator, authorizer);
-        // }
-
-        this.server = server;
-        metrics.setServer(server);
-        ScheduledExecutorService s = new ScheduledThreadPoolExecutor(1);
-        detectStart.startBrokerStartedDetection(port, s);
-    }
+    void startEmbeddedServer(Integer portParam, boolean secure, String username, String password,
+            @NonNull String persistenceFilenameParam) throws IOException;
 
     /**
      * Stops the embedded broker, if it is started.
      */
-    protected void stopEmbeddedServer() {
-        if (this.server != null) {
-            server.stopServer();
-            server = null;
-        }
-        detectStart.stopBrokerStartDetection();
-        metrics.setServer(null);
-    }
+    void stopEmbeddedServer();
 
     /**
-     * For testing: Returns true if the embedded server confirms that the MqttBrokerConnection is connected.
+     * Returns the MQTT broker connection, connected to the embedded broker
      */
-    protected boolean serverConfirmsEmbeddedClient() {
-        return server != null && server.getConnectionsManager().isConnected(CLIENTID);
-    }
-
-    @Override
-    public void connectionStateChanged(MqttConnectionState state, Throwable error) {
-        if (state == MqttConnectionState.CONNECTED) {
-            logger.debug("Embedded broker connection connected");
-        } else {
-            if (error == null) {
-                logger.debug("Offline - Reason unknown");
-            } else {
-                logger.debug(error.getMessage());
-            }
-        }
-
-        if (state != MqttConnectionState.CONNECTED && state != MqttConnectionState.CONNECTING) {
-            stopEmbeddedServer();
-        }
-    }
-
-    @Override
-    public void mqttEmbeddedBrokerStarted(boolean timeout) {
-        service.addBrokerConnection(CLIENTID, connection);
-
-        // Try a reconnect if not connected:
-        if (connection.connectionState() != MqttConnectionState.CONNECTED) {
-            try {
-                connection.start();
-            } catch (ConfigurationException | MqttException e) {
-                logger.debug(e.getLocalizedMessage());
-            }
-        } else {
-            connectionStateChanged(MqttConnectionState.CONNECTED, null);
-        }
-
-    }
-
-    @Override
-    public void connectedClientIDs(Collection<String> clientIDs) {
-        logger.debug("Connected clients: {}", clientIDs.stream().collect(Collectors.joining(", ")));
-    }
+    public MqttBrokerConnection getConnection();
```

Comment:
Methods of interface are implicit public, please remove it.

Reply:
A static check would be helpful to enforce this guideline

Reply:
You are right. Feel free to create a PR against the static-code-analyzer.

@svilenvul Can you integrate this one: https://pmd.github.io/pmd-6.2.0/pmd_rules_java_codestyle.html#unnecessarymodifier